tools/pipelines/test-perf-benchmarks.yml

```
@@ -70,160 +85,311 @@ stages:
+                tar -xvf $(Pipeline.Workspace)/test-files/$TAR_FILE.test-files.tar -C $(Pipeline.Workspace)/test-files
+                mv $(Pipeline.Workspace)/test-files/dist/test/* ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}/dist/test
+                mv $(Pipeline.Workspace)/test-files/src/test/* ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}/src/test
+
+          - task: Bash@3
+            displayName: 'Run execution-time tests'
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}
+              script: |
+                echo "Run execution-time tests for ${{ execTestPackage }}"
+                cp ${{ variables.testWorkspace }}/.npmrc . ;
+                npm i;
+                npm run test:benchmark:report;
+
+          - task: CopyFiles@2
+            displayName: 'Consolidate output files - execution time'
+            inputs:
+              sourceFolder: ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}/node_modules/@fluid-tools/benchmark/dist/.output/
+              contents: '**'
+              targetFolder: ${{ variables.testWorkspace }}/benchmarkOutput/${{ replace(replace(replace(execTestPackage, '@fluidframework/', '' ), '@fluid-', '' ), '/', '-') }}
+
+          - task: Bash@3
+            displayName: 'Cleanup package'
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}/node_modules/
+              script: |
+                echo "Cleanup package ${{ execTestPackage }} from ${{ variables.testWorkspace }}/node_modules/"
+                rm -rf ${{ execTestPackage }};
 
         - task: Bash@3
           displayName: 'Write measurements to Aria/Kusto - execution time'
           inputs:
             targetType: 'inline'
             workingDirectory: $(toolAbsolutePath)
             script: |
-              npm i;
-              npm run build:compile;
-              node --require @ff-internal/aria-logger bin/run --handlerModule $(toolAbsolutePath)/dist/handlers/executionTimeTestHandler.js --dir '../../benchmarkOutput';
+              echo "Write the following benchmark output to Aria/Kusto"
+              ls -la ${{ variables.testWorkspace }}/benchmarkOutput;
+              node --require @ff-internal/aria-logger bin/run --handlerModule $(toolAbsolutePath)/dist/handlers/executionTimeTestHandler.js --dir '${{ variables.testWorkspace }}/benchmarkOutput';
 
         - task: PublishPipelineArtifact@1
           displayName: 'Publish Artifact - Perf tests output - execution time'
           inputs:
-            targetPath: '${{ variables.buildDirectory }}/benchmarkOutput'
+            targetPath: '${{ variables.testWorkspace }}/benchmarkOutput'
             artifactName: 'perf-test-outputs_execution-time'
-          condition: succeededOrFailed()
+            condition: succeededOrFailed()
```

```suggestion
          condition: succeededOrFailed()
```

`condition` should be at the same level as `inputs`. This was actually a bug in the original pipeline, which @chentong7 fixed. Might be worth double checking this file to see if it got copy-pasted in more places.